### PR TITLE
ddl: remove limit of multi-partition drop/truncate

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Problem Summary: drop/truncate multi-partition is mainly implemented in #18419, this PR remove limit before that code.

### What is changed and how it works?

What's Changed:

remove length check and construct parameter list

How it Works:

mainly implemented in #18419

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 

- support drop multiple partitions

### CODE REVIEW

**Comment by lance6716** [comment]
PTAL @AilinKid @zimulala

File: ddl/partition.go [hunk 1]
```diff
@@ -603,16 +603,26 @@ func checkDropTablePartition(meta *model.TableInfo, partLowerNames []string) err
...
+				found = true
+				break
 			}
-			pids = append(pids, oldDefs[i].ID)
-			newDefs = append(oldDefs[:i], oldDefs[i+1:]...)
-			break
+		}
+		if found {
+			pids[k] = oldDefs[i].ID
+			k++
```
Comment (zimulala) [inline ddl/partition.go:622]: Why change the code？I think the original code(`pids = append(pids, oldDefs[i].ID)`) is more readable.

**Comment by zimulala** [comment]
PTAL @AilinKid @tangenta

File: ddl/partition.go (hunk 1 above)
Comment (lance6716) [reply ddl/partition.go]: addressed in https://github.com/pingcap/tidb/pull/18593/commits/1f8b3589c8740d153e116e91c81590140a0314e5

File: ddl/db_partition_test.go [hunk 2]
```diff
@@ -661,10 +661,11 @@ func (s *testIntegrationSuite5) TestAlterTableDropPartition(c *C) {
 	partition by range( hired ) (
 		partition p1 values less than (1991),
 		partition p2 values less than (1996),
-		partition p3 values less than (2001)
+		partition p3 values less than (2001),
+		partition p4 values less than (2006)
```
Comment (AilinKid) [inline ddl/db_partition_test.go:665]: beside this, you can write some dependent Test func about `drop multi partition and truncate multi partition`

File: ddl/partition.go [hunk 3]
```diff
@@ -603,16 +603,22 @@ func checkDropTablePartition(meta *model.TableInfo, partLowerNames []string) err
...
-	newDefs := make([]model.PartitionDefinition, 0, len(oldDefs)-1)
-	var pids []int64
-	for _, partName := range partLowerNames {
-		for i := 0; i < len(oldDefs); i++ {
-			if oldDefs[i].Name.L != partName {
-				continue
+	newDefs := make([]model.PartitionDefinition, 0, len(oldDefs)-len(partLowerNames))
+	pids := make([]int64, 0, len(partLowerNames))
+
+	// consider using a map to probe partLowerNames if too many partLowerNames
```
[the remark this answers is not in the document]
Comment (lance6716) [reply ddl/partition.go:609]: > this is a Todo?

I'm not sure, do you think we should use a threshold to turn to map/set probe? I think user will not have much partitions worth a overhead for map.

File: ddl/db_partition_test.go (hunk 2 above)
Comment (lance6716) [reply ddl/db_partition_test.go]: added in https://github.com/pingcap/tidb/pull/18593/commits/366ab3092be0d5a4f5c1b38661f0df8e280936e4

File: ddl/ddl_api.go [hunk 4]
```diff
@@ -2811,8 +2806,10 @@ func (d *ddl) DropTablePartition(ctx sessionctx.Context, ident ast.Ident, spec *
 		return errors.Trace(ErrPartitionMgmtOnNonpartitioned)
 	}
 
-	partName := spec.PartitionNames[0].L
-	partNames := []string{partName}
+	partNames := make([]string, len(spec.PartitionNames))
+	for i, partCIName := range spec.PartitionNames {
+		partNames[i] = partCIName.L
+	}
```
[the remark this answers is not in the document]
Comment (lance6716) [reply ddl/ddl_api.go:2809]: modify `TruncateTablePartition ` in https://github.com/pingcap/tidb/pull/18593/commits/366ab3092be0d5a4f5c1b38661f0df8e280936e4

File: ddl/partition.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (lance6716) [reply ddl/partition.go:609]: tested for about 10,000 partitions, nested for-loop begin to slow down. Should we switch to map/set probe on that threshold?

[the remark this answers is not in the document]
Comment (AilinKid) [reply ddl/partition.go:609]: Maybe user will not have so much partitions worth a overhead for map. It's ok here

**Comment by zimulala** [comment]
PTAL @tangenta

**Comment by lance6716** [comment]
ping @tangenta

**Comment by csuzhangxc** [comment]
ping @tangenta

**Comment by lance6716** [comment]
ping @tangenta

**Comment by tangenta** [comment]
@lance6716 Sorry for the late reply. :sweat_smile:

**Pull request merged**